scripts/in_progress/regrid.py

```diff
@@ -0,0 +1,405 @@
+    2. ds_out: output dataset (dataset that provides reference grid) with ds_type and input_grid added as attrs
+    3. m: regridding method* as string, patch as default
+    * Methods avaiable: https://xesmf.readthedocs.io/en/stable/user_api.html
+      Methods definition: https://earthsystemmodeling.org/regrid/#regridding-methods
+    **kwargs  keyword arguments
+    4. dir_wm: directory of weight matrix if not stored in current working directory
+    5. dir_wm_new: directory of newly create weight matrix, current working directory by default
+    Outputs:
+    1. Regridded xarray dataset/ dataarray
+    2. Regridder weight matrix.nc if missing
+    """
+    # build weight matrix filename
+    wm = ds_in.attrs['ds_type'] +'-'+ ds_in.attrs['input_grid'] + '_' + ds_out.attrs['ds_type'] + '-' + ds_out.attrs['input_grid'] + '_' + m + '.nc'
+    # build full path 
+    # if weight matrix is not stored in current working directory
+    if 'dir_wm' in kwargs.keys():
+        wm = os.path.join(kwargs['dir_wm'],wm)
+    else:
+        pass
+    # check if weight matrix is previously created
+    if os.path.isfile(wm):
+        print('weight matrix exists')
+        regridder = xe.Regridder(ds_in, ds_out, method = m, weights = wm)
+    # create weight matrix nc if missing
+    else:
+        print('building weight matrix...')
+        regridder = xe.Regridder(ds_in, ds_out, method = m)
+        # build xr dataset for export, adapted from xesmf/frontend.py (Line 759 - 768). DOI: https://doi.org/10.5281/zenodo.4294774
+        w = regridder.weights.data
+        dim = 'n_s'
+        wm_ds = xr.Dataset(
+            {'S': (dim, w.data), 'col': (dim, w.coords[1, :] + 1), 'row': (dim, w.coords[0, :] + 1)}
+        )
+        # add attrs
+        wm_ds.attrs['input_grid'] = ds_in.attrs['input_grid']
+        wm_ds.attrs['output_grid'] = ds_out.attrs['input_grid']
+        # writing weight matrix to disk
+        # for storing weight matrix to other location
+        if 'dir_wm_new' in kwargs.keys():
+            # check if folder exists
+            if os.path.exists(kwargs['dir_wm_new']) is False:
+                os.makedirs(kwargs['dir_wm_new'])
+            else:
+                pass
+            wm = os.path.join(kwargs['dir_wm_new'], wm)
+        wm_ds.to_netcdf(path = wm)
+        print('done')
+    # Perform regridding 
+    print('regridding...')
+    ds_in_re = regridder(ds_in, keep_attrs = True)
+    # Update grid attrs in regridded dataset
+    ds_in_re.attrs['output_grid'] = ds_out.attrs['input_grid']
+    print('done')
+    return ds_in_re
+
+# Interpolate to WRF24 grids
+def to_WRF_grid(ds, wrf_proj):
+    """
+    Function: Interpolate input dataset (WRF/ERA5/CERES/Rutgers) to WRF24 grid and perform remapping 
+              remap from polar to WRF24 for i.ie. Rutgers Northern Hemisphere 24 km Weekly Snow Cover Extent
+              or from WGS1984 to WRF24 for i.e. ERA5 Land, ERA5 plev, ERA5 slev
```

Comment:
This function is necessary, because we are dealing with a rotated pole with WRF, is it? I can see how this works with the Rutgers dataset or any dataset where lat/lon are 2D arrays, but in cases like ERA5, where lat/lon are just vectors, I am not sure this works... can you explain?

Reply:
![image](https://github.com/user-attachments/assets/fa041ab3-7842-4586-be3a-cf3c57aee9b0)
I still encounter the rotated pole issue with any dataset regridded to WRF grid but it is resolved after changing to wrf coordinate system
![image](https://github.com/user-attachments/assets/fa956110-449f-4935-aa76-9c1a8e1ee84f)